modules/ROOT/pages/persistence-gateway.adoc

```diff
@@ -0,0 +1,122 @@
+
+=== Create a Custom Resource and Secret
+
+. Create a Kubernetes custom resource for your data store.
+.. Copy the custom resource template from <<Kubernetes Custom Resource Template>> to a file called `custom-resource.yaml`.
+.. Modify the fields of the template as required for your environment.
+.. Run `kubectl apply -f custom-resource.yaml` to add the custom resource to your cluster.
+. Create a Kubernetes secret.
+.. Copy the Kubernetes secret template from <<Kubernetes Secret Template>> to a file called `kubernetes-secret.yaml`.
+.. Modify the fields of the template to include connectivity information and credentials for your installation.
+.. Run `kubectl apply -f kubernetes-secret.yaml`.
+
+=== Kubernetes Custom Resource Template
+
+Use the example template below to create a Kubernetes custom resource. This custom resource enables the Kubernetes cluster to connect to your data store.
+
+----
+apiVersion: rtf.mulesoft.com/v1
+kind: PersistenceGateway
+metadata:
+	name: persistent-gateway-config
+	namespace: rtf
+spec:
+	objectStore:
+    	backendDriver: postgresql
+    	maxBackendConnectionPool: 10
+		maxKeysPerStore: 5000
+    	maxStores: 4
+    	replicas: 1
+		resources:
+			limits:
+				cpu: 1m
+        		memory: 200Mi
+			requests:
+				cpu: 0.5m
+				memory: 100Mi
+		secretRef:
+			name: persistence-gateway-creds
+----
+
+Below are descriptions of the fields of the template you modify:
+
+[%header,cols="3*a"]
+|===
+| Field | Description | Default Value
+| `backendDriver` | Specifies the driver used by the data store. Only `postgresql` is supported. | `postgresql`
+| `maxBackendConnectionPool` | | 10
+| `maxKeysPerStore` | | 5000
+| `maxStores` | | 5000
+| `replicas` | | 5000
+| `resources: limits: cpu` | | 1m
+| `resources: limits: memory` | | 200Mi
+| `resources: requests: cpu` |  | 0.5m
+| `resources: requests: memory` | | 100Mi
+| `secretRef: name:` | Specifies the name of the persistence gateway credentials defined in the Kubernetes secret file.| `persistence-gateway-creds`
+|===
+
+=== Kubernetes Secret Template
+
+Use the example template below to create a Kubernetes secret. The Kubernetes cluster uses this secret to connect to your data store.
+
+----
+apiVersion: v1
+data:
+ persistence-gateway-creds: cG9zdGdyZXNxbDovL3Bvc3RncmVzOlRlc3RpbmdQb3N0Z3Jlc0BhdXRvbWF0aW9uLXRlc3RpbmctcnYuY2ZqcHFoeTZlYWhzLnVzLWVhc3QtMS5yZHMuYW1hem9uYXdzLmNvbTo1NDMyL1JURg==
+kind: Secret
+metadata:
+  name: persistence-gateway-creds
+  namespace: rtf
+type: Opaque
+----
+
+Below are descriptions of the fields of the template you should modify:
+
+[%header,cols="2*a"]
+|===
+| Field | Description
+| `metadata: persistence-gateway-creds` |
+| `metadata: namespace` | The internal identifier for this secret. This name must match the value of the `secretRef:name` field of your Kubernetes resource.
+| `type` | Specifies the type of Kubernetes key. Only `Opaque` is supported.
+|===
```

Comment:
We also need to add a NOTE highlighting few things:
1. The data key of the secret should be `persistence-gateway-creds`
2. The secret needs to applied first and then refer the secret inside the custom resource before applying the custom resource
3. The custom resource name has to be `default` and any other `PersistenceGateway` type customer resources will be ignored.